### PR TITLE
Add kubectl alpha apply-manifest command

### PR DESCRIPTION
**Pull request by mengqiy**
This PR is base on #52570. This PR will need to update when we decide the types in #52570.

This PR add a new command: `kubectl alpha expand`
It will generate API resources that can be piped to `kubectl apply|create|delete|replace|... -f`.
This PR is a prototype of [proposal](https://docs.google.com/document/d/1cLPGweVEYrVqQvBLJg6sxV-TrE5Rm2MNOBA_cxZP2WU/edit#).
What is supported:
- inject labels (TODO: seems we will need to update selector as well)
- inject annotations
- add prefix to the resource name
- apply overlay on top of base

What is not supported (need to discuss what the behavior should be):
- configMap injection
- secret injection

```release-note
TBD
```

/assign @monopole 
cc: @bgrant0607 @pwittrock

### CODE REVIEW

**Comment by mengqiy** [comment]
Will add some tests shortly.

File: pkg/kubectl/cmd/manifest/apply_manifest.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	return nil
+}
+
+// Complete completes apply-manifest command.
+func Complete(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+// RunApplyManifest runs apply-manifest command (do real work).
+func RunApplyManifest(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, options *applyManifestOptions) error {
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/manifest/apply_manifest.go]: I would create an expand command that generates all the resources. The user could then decide whether to apply, create, get, delete, whatever.

Comment (mengqiy) [reply pkg/kubectl/cmd/manifest/apply_manifest.go]: > I would create an expand command that generates all the resources. The user could then decide whether to apply, create, get, delete, whatever.

This is what this command does now.
I need to change the command name. Sorry for the confusion.

**Comment by smarterclayton** [comment]
This should be a binary extension command, not a new in tree command.  Until the manifest format proves its value, i don't think it belongs in tree.

**Comment by smarterclayton** [comment]
I certainly won't be comfortable with this in tree until it's shown it improves workflows over helm examples / solves the problems equally as well.

**Comment by monopole** [comment]
Hey Mengqiy - we're discussing which repo to move this to (other than kubernetes/kubernetes).

**Comment by bgrant0607** [comment]
@lavalamp @pwittrock Is this PR still useful?

**Pull request closed without merging**